File: Nivedita/Array/Level2/First Repeating Element .cpp
class Solution {
  public:
    // Function to return the position of the first repeating element.
    int firstRepeated(int arr[], int n) {
       map<int,pair<int,bool>>mp;
       int j=INT_MAX;
       for(int i=0;i<n;i++)
       {
           if(mp[arr[i]].second==true)
           {
              j=min(j,mp[arr[i]].first); 
           }
           mp[arr[i]]={i+1,true};
           
       }
       if(j<n)
       {
       return j;
       }
       return -1;// code here
    }
};

File: JyotiKumari/Array(GFG)/First repeating element.py
from collections import OrderedDict

class Solution:
    #Function to return the position of the first repeating element.
    def firstRepeated(self,arr, n):
        
        #arr : given array
        #n : size of the array
        d = OrderedDict()
        for element in arr:
            d[element] = 0
        for element in arr:
            d[element]+=1
        index = 1
        for element in d:
            if(d[element]>1):
                return index
            index+=1
        return -1